core/arch/arm/kernel/generic_boot.c

```diff
@@ -552,6 +555,37 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_pseudo_device_node(void *fdt)
+{
+	char s[33];
+	int offs, n = 0, m, len;
+	const uint32_t *p;
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "pseudo-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(pseudo_device_id))
```

Comment:
Yeah.. len is forced to be an int by libfdt api.  And it wants to do a comparison against unsigned (size_t) sizeof.